.github/buildomat/build-and-test.sh

```diff
@@ -112,7 +112,13 @@ export RUSTC_BOOTSTRAP=1
 
 # Build all the packages and tests, and keep track of how long each took to build.
 # We report build progress to stderr, and the "--timings=json" output goes to stdout.
-ptime -m cargo build -Z unstable-options --timings=json --workspace --tests --locked --verbose 1> "$OUTPUT_DIR/crate-build-timings.json"
+#
+# The build graph ends up building several bin/test targets that depend on
+# omicron-nexus at the same time, which uses significant memory to compile on
+# illumos. To mitigate this we build everything except omicron-nexus's bin/test
+# targets first, then finish the build after.
+ptime -m cargo build -Z unstable-options --timings=json --workspace --exclude=omicron-nexus --tests --locked --verbose 1>> "$OUTPUT_DIR/crate-build-timings.json"
+ptime -m cargo build -Z unstable-options --timings=json --workspace --tests --locked --verbose 1>> "$OUTPUT_DIR/crate-build-timings.json"
```

Comment:
take it or leave it: these are pretty lengthy, i might consider making them multiline...